types/mempool/mempool.go

```diff
@@ -1,31 +1,41 @@
-package types
+package mempool
 
-// MempoolTx we define an app-side mempool transaction interface that is as
+import (
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+// Tx defines an app-side mempool transaction interface that is as
 // minimal as possible, only requiring applications to define the size of the
-// transaction to be used when reaping and getting the transaction itself.
+// transaction to be used when inserting, selecting, and deleting the transaction.
 // Interface type casting can be used in the actual app-side mempool implementation.
-type MempoolTx interface {
-	Tx
+type Tx interface {
```

Comment:
Let's be sure these changes are captured in the ADR (including a changelog in the ADR)